service.impl/src/test/java/uni/miskolc/ips/ilona/tracking/service/impl/ZoneQueryServiceTest.java

```diff
@@ -0,0 +1,45 @@
+package uni.miskolc.ips.ilona.tracking.service.impl;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import uni.miskolc.ips.ilona.measurement.controller.dto.ZoneDTO;
+import uni.miskolc.ips.ilona.tracking.service.gateway.ZoneQueryService;
+import uni.miskolc.ips.ilona.tracking.service.gateway.ZoneQueryServiceSIConfig;
+
+import java.util.Collection;
+
+public class ZoneQueryServiceTest {
```

Comment:
ApplicationContext could be loaded here with annotation so it shouldn't be instantiated in each test methods. Moreover, Before method would be better to load the same application context before each test methods.